File: example/src/main/java/com/github/simonpercic/oklogexample/ui/MainMvp.java
package com.github.simonpercic.oklogexample.ui;

/**
 * Main MVP.
 *
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
interface MainMvp {

    interface View {

        void enableButtons();

        void disableButtons();

        void showMessage(String message);
    }

    interface Presenter {

        void onBtnGetClicked();

        void onBtnPostClicked();

        void onBtnPutClicked();

        void onBtnDeleteClicked();

        void onBtnHeaderClicked();

        void onDestroy();
    }
}

File: example/src/main/kotlin/com/github/simonpercic/oklogexample/ui/MainMvp.kt
package com.github.simonpercic.oklogexample.ui

/**
 * Main MVP.

 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
internal interface MainMvp {

    interface View {

        fun enableButtons()

        fun disableButtons()

        fun showMessage(message: String)
    }

    interface Presenter {

        fun onBtnGetClicked()

        fun onBtnPostClicked()

        fun onBtnPutClicked()

        fun onBtnDeleteClicked()

        fun onBtnHeaderClicked()

        fun onDestroy()
    }
}
